api/src/main/java/jakarta/data/metamodel/restrict/Restrict.java

```diff
@@ -142,109 +122,48 @@ public static <T> Restriction<T> not(Restriction<T> restriction) {
-    public static <T> UnaryRestriction<T> notNull(String attribute) {
-        return new UnaryRestrictionRecord<>(attribute, UnaryOperator.IS_NOT_NULL);
+    public static <T> BasicRestriction<T> notNull(String attribute) {
+        return new BasicRestrictionRecord<>(attribute, Constraint.isNull(), true);
     }
 
     public static <T> TextRestriction<T> notContains(String substring, String attribute) {
-        String pattern = toLikeEscaped(CHAR_WILDCARD, STRING_WILDCARD, true, substring, true);
-        return new TextRestrictionRecord<>(attribute, Operator.NOT_LIKE, ESCAPED, pattern);
+        return new TextRestrictionRecord<>(attribute, Like.substring(substring), true);
     }
 
-    public static <T> TextRestriction<T> notEndsWith(String suffix, String attribute) {
-        String pattern = toLikeEscaped(CHAR_WILDCARD, STRING_WILDCARD, true, suffix, false);
-        return new TextRestrictionRecord<>(attribute, Operator.NOT_LIKE, ESCAPED, pattern);
+    public static <T> TextRestriction<T> notStartsWith(String prefix, String attribute) {
+        return new TextRestrictionRecord<>(attribute, Like.prefix(prefix), true);
     }
 
-    public static <T> Restriction<T> notIn(Set<?> values, String attribute) {
-        return new BasicRestrictionRecord<>(attribute, Operator.NOT_IN, values);
+    public static <T> TextRestriction<T> notEndsWith(String suffix, String attribute) {
+        return new TextRestrictionRecord<>(attribute, Like.suffix(suffix), true);
     }
 
-    public static <T> TextRestriction<T> notLike(String pattern, String attribute) {
-        return new TextRestrictionRecord<>(attribute, Operator.NOT_LIKE, pattern);
+    public static <T> BasicRestriction<T> notIn(Set<?> values, String attribute) {
+        return new BasicRestrictionRecord<>(attribute, Constraint.in(values), true);
     }
 
-    public static <T> TextRestriction<T> notLike(String pattern,
-                                                  char charWildcard,
-                                                  char stringWildcard,
-                                                  String attribute) {
-        String p = toLikeEscaped(charWildcard, stringWildcard, false, pattern, false);
-        return new TextRestrictionRecord<>(attribute, Operator.NOT_LIKE, ESCAPED, p);
+    public static <T> TextRestriction<T> notLike(String pattern, String attribute) {
+        return new TextRestrictionRecord<>(attribute, Like.pattern(pattern), true);
     }
 
-    public static <T> TextRestriction<T> notStartsWith(String prefix, String attribute) {
-        String pattern = toLikeEscaped(CHAR_WILDCARD, STRING_WILDCARD, false, prefix, true);
-        return new TextRestrictionRecord<>(attribute, Operator.NOT_LIKE, ESCAPED, pattern);
+    public static <T> TextRestriction<T> notLike(String pattern, char charWildcard, char stringWildcard, String attribute) {
+        return new TextRestrictionRecord<>(attribute, Like.pattern(pattern, charWildcard, stringWildcard), true);
```

Comment:
`notLike` should match `like` with respect to escape character
```suggestion
    public static <T> TextRestriction<T> notLike(String pattern, char charWildcard, char stringWildcard, char escape, String attribute) {
        return new TextRestrictionRecord<>(attribute, Like.pattern(pattern, charWildcard, stringWildcard, escape), true);
```

Reply:
actually it's an overload.